### PR TITLE
fix: issue with long inline links not breaking

### LINKED ISSUE #3110: Inline links should wrap
Links used inside a paragraph should wrap.

In case of `data-variant="inline"` the  property should change to `white-space: wrap;` and the display should change to `   display: inline;`

https://github.com/db-ui/mono/blob/cc84e5d56beaab44a5fa7a3fc5fd144e8b843769/packages/components/src/styles/_link-components.scss#L43

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #3110

## Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/components/link/link.scss [hunk 1]
```diff
@@ -17,6 +17,11 @@
 		@extend %db-link-height;
 	}
 
+	&[data-variant="inline"] {
+		white-space: inherit;
+		display: inline;
+	}
+
```
Comment (mfranzke) [inline packages/components/src/components/link/link.scss:24]: What do you think about wrapping the following line instead by a selector that makes this more specific w/o the need to rewrite it afterwards?
https://github.com/db-ui/mono/blob/main/packages/components/src/styles/_link-components.scss#L43

Wrapping it with:
``` CSS
&:not([data-variant="inline"]) {
```

Comment (nmerget) [reply packages/components/src/components/link/link.scss]: done

File: showcases/react-showcase/src/components/form/index.tsx
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:1]: isn't it that we would need to include this components variant on our regular documentation page as well?

Comment (mfranzke) [reply showcases/react-showcase/src/components/form/index.tsx:1]: https://github.com/db-ui/mono/issues/3237

**Review by mfranzke — APPROVED** [review]

File: showcases/react-showcase/src/components/form/index.tsx
Comment (nmerget) [reply showcases/react-showcase/src/components/form/index.tsx:1]: It's not in the design specs. We need to discuss this

**Pull request merged**